source/elements/oneDAL/source/data_management/index.rst

```diff
@@ -126,113 +130,93 @@ used at the data preparation and data processing stages for the following:
   format`, e.g. by blocks with the defined :capterm:`data layout`.
 
-For thread-safety reasons and better integration with external entities, a table
-provides a read-only access to the data within it, thus, table concept
-implementations shall be :capterm:`immutable <immutability>`.
+.. note::
+  For thread-safety reasons and better integration with external entities, a table
+  provides a read-only access to the data within it, thus, table concept
+  implementations shall be :capterm:`immutable <immutability>`.
+
+.. warning::
+  Table can represent not a datasets only - it can be a matrix, a vector (if
+  number of columns :math:`p = 1`), even a single value.
+
+  When table represent a dataset, the table rows represent dataset observations,
+  and columns represent features according to the :txtref:`dataset` definition.
 
 This concept has different logical organization and physical :capterm:`format of
 the data <data format>`:
 
-- Logically, a table is a :txtref:`dataset` with :math:`n` rows and
-  :math:`p` columns. Each row represents an :capterm:`observation` and each
-  column is a :capterm:`feature` of a dataset. Physical amount of bytes needed
-  to store the data differ from the number of elements :math:`n \times p` within
-  a table.
+- Logically, a table contains :math:`n` rows and :math:`p` columns.
+  Every column may have its own type of data values and set of allowed
+  operations on them.
 
 - Physically, a table can be organized in different ways: as a
   :capterm:`homogeneous <homogeneous data>`, :capterm:`contiguous <contiguous
   data>` array of bytes, as a :capterm:`heterogeneous <heterogeneous data>` list
   of arrays of different :capterm:`data types <data type>`, in a
-  compressed-sparse-row format.
+  compressed-sparse-row format. Physical amount of bytes needed to store the
+  data differ from the number of elements :math:`n \times p` within a table.
 
 For details, see :txtref:`tables` section.
 
-.. _metadata:
-
-Metadata
---------
-
-Metadata concept is associated with a :txtref:`dataset` and holds information
-about its structure and type. This information shall be enough to determine the
-particular type of a dataset, and it helps to understand how to interact with a
-dataset in |dal_short_name| (for example, how to use it at a particular stage of
-:txtref:`data analytics pipeline <data_analytics_pipeline>` or how to access its
-data).
-
-For each dataset, its metadata shall contain:
-
-- The number of rows :math:`n` and columns :math:`p` in a dataset.
+.. _table_metadata:
 
-- The type of each :capterm:`feature` (e.g. :capterm:`nominal <nominal
-  feature>`, :capterm:`interval <interval feature>`).
+Table metadata
+--------------
 
-- The :capterm:`data type` of each feature (e.g. :code:`float` or
-  :code:`double`).
+Table metadata concept contains an additional information about data into the
+table, that is not included into the table itself but can be useful in the
+particular use case.
```

Comment:
Can be shorter, w/o duplication of words:
Table metadata...into the table:
1. data types of table columns
2....

'When we speak... ' we need a different a bit more formal wording in the specs. We probably want to say that those logical types apply both to cases of data and data processing. Do we need to talk about it in this case?

Reply:
Rewritten. About logical types - I want to highlight because these types are bound with "feature" term, which applies only for datasets, not matrices. That's why I want to say that it can be generalized on non-dataset cases. Or we need to correct the terms of ordinal/nominal/... features?